platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/network/connectivity/SocketFactoryTest.java

```diff
@@ -154,17 +172,47 @@ void tlsFactoryTest(final AddressBook addressBook, final Map<NodeId, KeysAndCert
         final KeysAndCerts keysAndCerts1 = keysAndCerts.get(node1);
         final KeysAndCerts keysAndCerts2 = keysAndCerts.get(node2);
 
+        final Address address1 = addressBook.getAddress(node1);
+        final Address address2 = addressBook.getAddress(node2);
+        final PeerInfo peer1 = new PeerInfo(
+                node1,
+                address1.getSelfName(),
+                Objects.requireNonNull(address1.getHostnameExternal()),
+                Objects.requireNonNull(address1.getSigCert()));
+        final PeerInfo peer2 = new PeerInfo(
+                node2,
+                address2.getSelfName(),
+                Objects.requireNonNull(address2.getHostnameExternal()),
+                Objects.requireNonNull(address2.getSigCert()));
+        final List<PeerInfo> peerInfoList = List.of(peer1, peer2);
+
         testSocketsBoth(
                 NetworkUtils.createSocketFactory(node1, addressBook, keysAndCerts1, TLS_NO_IP_TOS_CONFIG),
-                NetworkUtils.createSocketFactory(node2, addressBook, keysAndCerts2, TLS_NO_IP_TOS_CONFIG));
+                NetworkUtils.createSocketFactory(node2, addressBook, keysAndCerts2, TLS_NO_IP_TOS_CONFIG),
+                peerInfoList);
         testSocketsBoth(
                 NetworkUtils.createSocketFactory(node1, addressBook, keysAndCerts1, TLS_IP_TOS_CONFIG),
-                NetworkUtils.createSocketFactory(node2, addressBook, keysAndCerts2, TLS_IP_TOS_CONFIG));
+                NetworkUtils.createSocketFactory(node2, addressBook, keysAndCerts2, TLS_IP_TOS_CONFIG),
+                peerInfoList);
+
+        final PeerInfo peer3 = new PeerInfo(
+                node1,
+                address1.getSelfName(),
+                Objects.requireNonNull(address1.getHostnameExternal()),
+                mock(X509Certificate.class));
+        Mockito.when(((X509Certificate) peer3.signingCertificate()).getSubjectX500Principal())
+                .thenReturn(mock(X500Principal.class));
+
+        final List<PeerInfo> peerInfoListNonMatchingCertPeers = List.of(peer3);
+        testSocketsBoth(
```

Comment:
I'm finding this test logic difficult to follow, with `testSocketsBoth` asserting outcome of `validateTLSPeer` based on what is or isn't mocked. Maybe there is a way to simplify these assertions / make it clear what the expected outcome is

Reply:
Maybe having 2 versions of `testSockets` might make sense: 1 that asserts successful validation, 1 unsuccessful. Or `testSockets` could be broken into separate pieces so that there isn't any duplication.

At a minimum, something to aid understanding would be adding a comment or 2 explaining the logic of checking if an item is mocked

Reply:
Added a comment.